src/librpmem/rpmem.c

```diff
@@ -620,7 +624,32 @@ rpmem_persist(RPMEMpool *rpp, size_t offset, size_t length, unsigned lane)
 		return -1;
 	}
 
-	int ret = rpmem_fip_persist(rpp->fip, offset, length, lane);
+	int ret = 0;
+
+	if (progress_cb == NULL) {
+		ret = rpmem_fip_persist(rpp->fip, offset, length, lane);
+	} else {
+		if (msg == NULL || *msg == '\0')
+			msg = "Persisting data";
+
+		size_t off = 0;
+		size_t next_off = 0;
+
+		progress_cb(msg, 0, length);
+		for (unsigned i = 0; i < 100; ++i) {
```

Comment:
You are doing this often enough to consider wrapping it in a helper function/macro.